### ISSUE TITLE
Dasel fails to parse a "null" yaml file

### ISSUE THREAD

**Issue by varac** [TomWright/dasel#99]
**Describe the bug**

Dasel fails to parse yaml file with `null` as the only content.

**To Reproduce**

1. `echo 'null' > /tmp/null.yml`
2. `echo '---' > /tmp/null2.yml`
3. `touch /tmp/null3.yml`
4. `dasel -f /tmp/null.yml .`
5. repeat with other files
6. All cmds fail with `panic: reflect: call of reflect.Value.Interface on zero Value`

**Expected behavior**

Dasel should not fail on valid (but empty) yaml.

**Desktop (please complete the following information):**
 - OS: Ubuntu 20.10
 - Version: v1.13.2

**Additional context**

`yamllint`, `yq` or `ytt` all don't fail on the same input files.

**Comment by TomWright** [comment]
Thank you for the clear examples. I'll take a look at this soon 👍

**Comment by TomWright** [comment]
Hi @varac,

I've made the changes required to get this working as expected. The fixes should also apply to other formats.

The one strange thing I've noticed is that empty YAML documents are marshal'd as JSON - this is down to the YAML package being used and I have raised an issue here: https://github.com/go-yaml/yaml/issues/708

```bash
echo 'null' | go dasel -p yaml .
{}
echo '---' | go dasel -p yaml .
{}
echo '' | dasel -p yaml . 
{}
```

**Comment by varac** [comment]
Thanks!

**Comment by TomWright** [comment]
It seems that `{}` is actually the correct representation: https://stackoverflow.com/questions/33510094/syntax-for-empty-dictionary-in-yaml

I'm happy to merge this in and get it released now.

**Comment by TomWright** [comment]
This is now released under `v1.13.3`.
